src/InterventionBackend.php

```diff
@@ -636,36 +714,67 @@ protected function createCloneWithResource($resourceOrTransformation)
     }
 
     /**
-     * Declare this image is being manipulated
+     * Clear any cached errors / metadata for this image
      *
-     * @param string $hash Hash of original file being manipluated
-     * @return bool
+     * @param string $hash
+     * @param string $variant
      */
-    protected function markStart($hash)
+    protected function markSuccess($hash, $variant)
     {
-        return $this->getCache()->set(self::CACHE_MARK.$hash, 1);
+        $key = $this->getErrorCacheKey($hash, $variant);
+        $this->getCache()->deleteMultiple([
+            $key.'_reason',
+            $key.'_ttl'
+        ]);
     }
 
     /**
-     * Declare that this image has been successfully manipulated
+     * Mark this image as failed to load
      *
-     * @param string $hash
-     * @return bool
+     * @param string $hash Hash of original file being manipluated
+     * @param string $variant Variant being loaded
+     * @param string $reason Reason this file is failed
      */
-    protected function markEnd($hash)
+    protected function markFailed($hash, $variant, $reason = self::FAILED_UNKNOWN)
     {
-        return $this->getCache()->delete(self::CACHE_MARK.$hash);
+        $key = $this->getErrorCacheKey($hash, $variant);
+
+        // Get TTL for error
+        $errorTTLs = $this->config()->get('error_cache_ttl');
+        $ttl = isset($errorTTLs[$reason]) ? $errorTTLs[$reason] : $errorTTLs[self::FAILED_UNKNOWN];
+
+        // Detect increasing waits
+        if (is_string($ttl) && strstr($ttl, ',')) {
+            $ttl = preg_split('#\s*,\s*#', $ttl);
+        }
+        if (is_array($ttl)) {
+            $index = min(
+                $this->getCache()->get($key.'_ttl', -1) + 1,
+                count($ttl) - 1
+            );
+            $this->getCache()->set($key.'_ttl', $index);
+            $ttl = $ttl[$index];
+        }
+        if (!is_numeric($ttl)) {
+            throw new LogicException("Invalid TTL {$ttl}");
+        }
+        // Treat 0 as unlimited
+        $ttl = $ttl ? (int)$ttl : null;
+        $this->getCache()->set($key.'_reason', $reason, $ttl);
     }
 
     /**
-     * Determine if this file has been loaded for manipulation, but could not be resized
+     * Determine reason this file could not be loaded.
+     * Will return one of the FAILED_* constant values, or null if not failed
      *
      * @param string $hash Hash of the original file being manipulated
-     * @return bool
+     * @param null $variant
+     * @return string|null
      */
-    protected function hasFailed($hash)
+    protected function hasFailed($hash, $variant)
```

Comment:
maybe make `$variant` optional with null as default?
I've noticed the docblock doesn't match the string it expects and this being called with `$variant = null` occasionally.